contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -0,0 +1,94 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.python.interpreter_cache import PythonInterpreter, PythonInterpreterCache
+from pants.backend.python.python_requirement import PythonRequirement
+from pants.backend.python.targets.python_binary import PythonBinary
+from pants.backend.python.targets.python_library import PythonLibrary
+from pants.backend.python.targets.python_tests import PythonTests
+from pants.backend.python.tasks.python_task import PythonTask
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnit, WorkUnitLabel
+from pants.util.contextutil import temporary_dir
+from pants.util.process_handler import subprocess
+from pex.pex import PEX
+from pex.pex_builder import PEXBuilder
+from pex.pex_info import PexInfo
+
+
+class MypyTask(PythonTask):
+  """Invoke the mypy static type analzyer for Python."""
+
+  _PYTHON_SOURCE_EXTENSION = '.py'
+
+  @classmethod
+  def supports_passthru_args(cls):
+    return True
+
+  def __init__(self, *args, **kwargs):
+    super(MypyTask, self).__init__(*args, **kwargs)
+
+  def py3_interpreter(self):
+    interpreters = self._interpreter_cache.setup(filters=['CPython>3'])
+    return min(interpreters) if interpreters else None
+
+  @staticmethod
+  def is_non_synthetic_python_target(target):
+    return not target.is_synthetic and isinstance(target, (PythonLibrary, PythonBinary, PythonTests))
+
+  def _calculate_python_sources(self, targets):
+    """Generate a set of source files from the given targets."""
+    python_eval_targets = filter(self.is_non_synthetic_python_target, targets)
+    sources = set()
+    for target in python_eval_targets:
+      sources.update(
+        source for source in target.sources_relative_to_buildroot()
+        if os.path.splitext(source)[1] == self._PYTHON_SOURCE_EXTENSION
+      )
+    return list(sources)
+
+  def _run_mypy(self, py3_interpreter, mypy_args, **kwargs):
+    pex_info = PexInfo.default()
+    pex_info.entry_point = 'mypy'
+    chroot = self.cached_chroot(interpreter=py3_interpreter, pex_info=pex_info, targets=[],
+      extra_requirements=[PythonRequirement('mypy==0.550')])
```

Comment:
this probably deserves either an option to override the version (e.g. `--mypy-version` w `default='0.550'`) or a tool address (e.g. `--mypy-tool-address` w `default='3rdparty/python:mypy'`).

Reply:
Done.